src/qibo/quantum_info/metrics.py

```diff
@@ -590,6 +590,59 @@ def bures_distance(state, target, check_hermitian: bool = False, backend=None):
     return distance
 
 
+def entanglement_fidelity(channel, nqubits: int, state=None, backend=None):
+    """Entanglement fidelity of a ``channel`` :math:`\\mathcal{E}` on ``state``
+    :math:`\rho`, which is given by
+
+    .. math::
+        \\begin{align*}
+          F_{\\mathcal{E}} &= \\text{fidelity}(\\rho_{f}, \\rho) \\nonumber \\\\
+          &= \\tr(\\rho_{f} \\, \\rho)
+
+    where
+
+    .. math::
+        \\rho_{f} = \\mathcal{E}_{A} \\otimes I_{B}(\\rho)
+
+    is the state after the channel :math:`\\mathcal{E}` was applied to
+    partition :math:`A`.
+
+    Args:
+        channel (:class:`qibo.gates.channels.Channel`): quantum channel
+            acting on partition :math:`A`.
+        nqubits (int): total number of qubits in ``state``.
+        state (ndarray, optional): statevector or density matrix to be evolved
+            by ``channel``. If ``None``, defaults to the maximally entangled state
+            :math:`\\frac{1}{2^{n}} \\, \\sum_{k} \\, \\ket{k}\\ket{k}`, where
+            :math:`n` is ``nqubits``. Defaults to ``None``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses :class:`qibo.backends.GlobalBackend`.
+            Defaults to ``None``.
+
+    Returns:
+        float: Entanglement fidelity :math:`F_{\\mathcal{E}}`.
+    """
+    if backend is None:
+        backend = GlobalBackend()
+
+    dim = 2**nqubits
+
+    density_matrix = True if state is not None and len(state.shape) == 2 else False
+
+    if state is None:
+        state = np.ones(dim, dtype=complex) / np.sqrt(dim)
+        state = backend.cast(state, dtype=state.dtype)
+
+    if density_matrix is True:
```

Comment:
Would it be possible to use `plus_state` here?
```suggestion
    if state is None:
        state = backend.plus_state(nqubits)

    if len(state.shape) == 2:
```